mm/kfence/kfence-test.c

```diff
@@ -422,6 +422,45 @@ static void test_memcache_ctor(struct kunit *test)
 	KUNIT_EXPECT_FALSE(test, report_available());
 }
 
+/* Test that memory is zeroed if requested. */
+static void test_gfpzero(struct kunit *test)
+{
+	const int size = PAGE_SIZE; /* PAGE_SIZE so we can use ALLOCATE_ANY. */
+	char *buf1, *buf2;
+	int i;
+
+	if (CONFIG_KFENCE_SAMPLE_RATE > 100) {
+		kunit_warn(test, "skipping ... would take too long\n");
+		return;
+	}
+
+	setup_test_cache(test, size, NULL);
+	buf1 = test_alloc(test, size, GFP_KERNEL, ALLOCATE_ANY);
+	for (i = 0; i < size; i++)
+		buf1[i] = i + 1;
+	test_free(buf1);
+
+	/* Try to get same address again -- this can take a while. */
+	for (i = 0;; i++) {
+		buf2 = test_alloc(test, size, GFP_KERNEL | __GFP_ZERO, ALLOCATE_ANY);
+		if (buf1 == buf2)
+			break;
+		test_free(buf2);
+
+		if (i == CONFIG_KFENCE_NUM_OBJECTS) {
```

Comment:
Hitting KFENCE_NUM_OBJECTS doesn't necessarily mean we cannot get the same object again.
But at least it's a reasonable stopping condition.

Reply:
It's an approximation. It might have also been allocated by system elsewhere and we're never getting it back. But at least this way we have cycled through the freelist at least once.

Reply:
Ok.